### ISSUE TITLE
Chat - Attachment from old dot chat shown empty rectangles in ND

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#44445]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.1-0 
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** rybkina+062524old@gmail.com
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/Web-Expensify/pull/42420
## Action Performed:
1. Sign up for gmail account and create collect policy in OD
2. Create report and send message and attachments
3. Tap on Concierge button > go to chat report

## Expected Result:
Attachment from old dot chat shown in ND with image

## Actual Result:
Attachment from old dot chat shown empty rectangles in ND

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/9ac0e4b8-baf8-41ae-8159-1ee8be5f0176

<img width="1397" alt="26 33" src="https://github.com/Expensify/App/assets/78819774/5bb0f641-9951-465f-9165-e1b81f0fe85a">

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017e4b531ab757c124</li>
        <li>Upwork Job ID: 1808449642258792691</li>
        <li>Last Price Increase: 2024-07-17</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @luacmartins</details>

**Comment by lanitochka17** [comment]
@twisterdotcom  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by twisterdotcom** [comment]
Sorry, I didn't get around to testing this again this week, my bad. Monday I will and get External on it I assume.

**Comment by twisterdotcom** [comment]
Okay, I can recreate, but I wonder if this is something that will just be resolved by sending the correct OD > ND onyx updates. Also, I didn't turn my Mic on for this, but I'm not redoing it.

https://github.com/Expensify/App/assets/9133401/217888e5-f567-4f21-b076-78f0dafd51b1

Asking internally: https://expensify.slack.com/archives/C03SSAQ7P/p1719851330833789

**Comment by twisterdotcom** [comment]
Okay, Onxy should be working for this, so going to make it external.

**Comment by amunim** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
attachments from old expensify cannot be previewed
### What is the root cause of that problem?
CORS issue:
![image](https://github.com/user-attachments/assets/90bc1d3d-e613-4e49-beab-cbbe3ed644a2)
### What changes do you think we should make in order to solve the problem?
We add a CORS header to the back-end to allow new.expensify.com
### What alternative solutions did you explore? (Optional)

**Comment by twisterdotcom** [comment]
How is this solution @dukenv0307?

**Comment by amunim** [comment]
@dukenv0307 new images are being loaded correctly, but old ones still same issue. Seeing that expensify and new.expensify are essentially same, we should add the header to the back-end to avoid any future CORS issues as well.
![image](https://github.com/user-attachments/assets/526d8f1d-d0cf-4cd0-b3c2-90e6d8fc9b13)
![image](https://github.com/user-attachments/assets/b11bafa9-5857-45e3-a4ec-caddf4772047)

**Comment by twisterdotcom** [comment]
@dukenv0307 what do you think? Do you agree? Do we need somebody internal on this?

**Comment by twisterdotcom** [comment]
This for sure isn't fixed:
<img width="2527" alt="image" src="https://github.com/user-attachments/assets/94ce2cb3-8c7d-4997-ae6a-ddb07eacccd3">

@amunim is correct that for old attachments, the issue remains.

**Comment by twisterdotcom** [comment]
>Access to fetch at '
https://www.expensify.com/chat-attachments/679266030899698153/w_43cfe77...jRg-1024.jRg' from origin'
https://new.expensify-com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the
requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource
with CORS disabled.

**Comment by amunim** [comment]
Does this mean we go with my proposal? (:|

**Comment by twisterdotcom** [comment]
Possibly, but implementation would be internal, so there would be no payment needed here.

**Comment by twisterdotcom** [comment]
Still require an internal engineer.

**Comment by twisterdotcom** [comment]
Gave this a retest - it's still weird that attachments sometimes load and sometimes appear to stall:

**Comment by twisterdotcom** [comment]
https://github.com/user-attachments/assets/359c9019-59ef-46dc-bc21-e31893b94837

**Comment by twisterdotcom** [comment]
Still no internal volunteer so far.

**Comment by mallenexpensify** [comment]
Reported last week too https://expensify.slack.com/archives/C049HHMV9SM/p1724428284702009

This seems like something we should prioritize, right? Possibly as part of #newdot-quality? 
Put another way.. If I were a paying user and I saw this bug, I wouldn't know why it's happening and would think NewDot is janky af if you can't upload/show an image in a chat in a chat app.

**Comment by twisterdotcom** [comment]
@muttmuure would you agree this is something we would pursue as part of NDQ?

**Comment by muttmuure** [comment]
Reading

**Comment by muttmuure** [comment]
Yeah let's do it. Some attachments work, some don't, so that's unreliable UX

**Comment by twisterdotcom** [comment]
Possibly fixed here: https://github.com/Expensify/Mobile-Expensify/pull/13071???

**Comment by muttmuure** [comment]
that would be greeeeaaaaat!

**Comment by luacmartins** [comment]
That's just HybridApp though. It seems like the issue happens on web, which is not affected by HybridApp

**Comment by twisterdotcom** [comment]
Do we think the resolution could be the same though?

**Comment by luacmartins** [comment]
I don't think so, we already pass the authtoken when requesting these receipts in App. The issue with App is that the browser itself if blocking the call because of CORS. Funny enough, the request goes through if you open the browser console, disable cache and refresh the page. Not sure why the first request is blocked.

**Comment by muttmuure** [comment]
Is this still happening?

**Comment by muttmuure** [comment]
How do we add a CORS header to the back end?

**Comment by luacmartins** [comment]
The correct headers seem to be added [here](https://github.com/Expensify/Salt/blob/e6fe16f1644fca3c951a9db59cb1e7a2b65f9ea1/shared_pkgs/nginx/files/conf.d/www.conf.template#L31-L42). I started [this discussion in Slack](https://expensify.slack.com/archives/C03TQ48KC/p1725651737370479) a couple of weeks ago, but we still didn't figure out what's causing it.

**Comment by justinpersaud** [comment]
Is anyone able to reliably reproduce? I'm trying to figure out where exactly the issue is coming from. I tried following the steps from @twisterdotcom's video but it shows up just fine for me:

<img width="818" alt="image" src="https://github.com/user-attachments/assets/6824d657-0900-4e98-928c-9b53f864ff5a">

Is the collect policy significant, since that is no longer an option?

**Comment by mallenexpensify** [comment]
I haven't seen it happen in a while (my repro steps were different than the one's in the OP)

**Comment by luacmartins** [comment]
I haven't seen it happen for a couple of week either.

**Comment by mallenexpensify** [comment]
Reported by @JmillsExpensify yesterday too
https://expensify.slack.com/archives/C049HHMV9SM/p1726661738595089
NewDot room link is [here](https://staging.new.expensify.com/r/1078638198657082/799203687895243501)

**Comment by twisterdotcom** [comment]
Confirmed I can recreate: https://expensify.slack.com/archives/C05LX9D6E07/p1727253691056909?thread_ts=1725992803.929979&cid=C05LX9D6E07

**Comment by twisterdotcom** [comment]
@luacmartins or @justinpersaud - any more thoughts on this?

**Comment by luacmartins** [comment]
FWIW I still can't find anything wrong with the headers. Also FWIW I can reproduce this issue on staging, but not on production.

https://github.com/user-attachments/assets/f131c44e-27ff-445a-a04c-a65ae2a61374

**Comment by luacmartins** [comment]
I'm also unable to reproduce on dev. So this seems to be affecting staging only. @twisterdotcom were you on production when you reproduced the issue?

**Comment by justinpersaud** [comment]
I put a PR up that I think may help here.

**Comment by luacmartins** [comment]
I can still reproduce the issue on staging 😞

**Comment by justinpersaud** [comment]
@luacmartins can you write out the full steps you're using?

**Comment by luacmartins** [comment]
These are the request headers

<img width="796" alt="Screenshot 2024-10-01 at 7 19 05 AM" src="https://github.com/user-attachments/assets/d3d039c4-aac2-449f-b901-85bf3defe8b8">

**Comment by justinpersaud** [comment]
Do you know if we changed anything with how we're fetching the image from the client in NewDot?

If you copy that request as cURL and paste it in your browser with all the headers, you will see the response contains the headers it says are missing:

```
< access-control-allow-origin: https://staging.new.expensify.com
< access-control-allow-credentials: true
```

**Comment by luacmartins** [comment]
I'm not aware of any changes to that. I'll ask in open-source. So you're saying that we're missing the `access-control-allow-origin` and `access-control-allow-credentials` headers in the request?

**Comment by luacmartins** [comment]
We could add those headers [here](https://github.com/Expensify/App/blob/9b4396bd2cc149cb3ee7d00ad901552030afebc4/src/components/Image/index.tsx#L54), but since I'm unable to reproduce in dev, that's a bit of a shot in the dark.

**Comment by luacmartins** [comment]
Yea, it's been pretty consistent for me, but only on staging (can't repro on dev nor prod)

**Comment by twisterdotcom** [comment]
Agree, this is only on staging for me too, so perhaps it's not super pressing.

**Comment by luacmartins** [comment]
Yea, this will only affect internal employees and contributors for now. We should still fix it, but maybe the priority is a bit lower since it doesn't affect customers

**Comment by mallenexpensify** [comment]
I don't remember seeing any/many bugs that are 'stuck' in staging and never hit production.  Is this something we need to document _somewhere_?   ie. in the main BZ SO, do we want to add something like "If a bug is stuck in staging and isn't affecting real-world users on production, feel free to treat it as lower priority'

**Comment by justinpersaud** [comment]
@luacmartins can you copy the request that is failing as curl or fetch and share it here? I tried repro steps again on staging, still can't reproduce it.

I need the full headers, request method and URL that is failing.

**Comment by luacmartins** [comment]
Ha now I can't reproduce using [these steps](https://github.com/Expensify/App/issues/44445#issuecomment-2385800574) anymore 🙃 

I did notice that now the first request to get the attachment is consistently cancelled and the 2nd one succeeds.

**Comment by justinpersaud** [comment]
For preflight requests, the first request is supposed to be an OPTIONS request and we respond back with a 204, and then it is supposed to request the actual image.

**Comment by luacmartins** [comment]
Closing as discussed [here](https://expensify.slack.com/archives/C05LX9D6E07/p1725992803929979)
